worker/changestream/eventqueue/eventqueue.go

```diff
@@ -101,8 +88,33 @@ func (s *subscription) Wait() error {
 func (s *subscription) loop() error {
 	defer close(s.changes)
 
-	<-s.tomb.Dying()
-	return tomb.ErrDying
+	select {
+	case <-s.tomb.Dying():
+		return tomb.ErrDying
+	case <-s.active:
+		return nil
+	}
+}
+
+// signal returns the channel that events will be sent on (signalled). It
+// checks that the subscription is not already dying, before passing back the
+// changes channel. This prevents a panic if you attempt to signal a dying
+// subscription.
+func (s *subscription) signal() chan<- changestream.ChangeEvent {
```

Comment:
I think this method should take an event and a dying channel and return nothing. The problem with returning the s.changes channel is that it still leaves us open to the eventqueue loop soft locking on the changes channel of an unsubscribing subscription.

Reply:
Yes, I see it.